src/rust/engine/options/src/parse.rs

```diff
@@ -14,16 +14,36 @@ peg::parser! {
         rule value<T>(parse_value: rule<T>) -> T
             = whitespace()* value:parse_value() whitespace()* { value }
 
-        rule integer() -> i64
-            = i:$("-"?['0'..='9']+) { i.parse::<i64>().unwrap() }
+        rule false() -> bool
+            = ("F"/"f") ("A"/"a") ("L"/"l") ("S"/"s") ("E"/"e") { false }
+
+        rule true() -> bool
+            = ("T"/"t") ("R"/"r") ("U"/"u") ("E"/"e") { true }
+
+        pub(crate) rule bool() -> bool
+            = b:(true() / false()) { b }
+
+        // Python numeric literals can include digit-separator underscores. It's unlikely
+        // that anyone relies on those in option values, but since the old Python options
+        // system accepted them, we support them here.
```

Comment:
The grammar was lifted directly from https://docs.python.org/3/library/functions.html#float

Reply:
But of course the proof is in the tests